### ISSUE TITLE
Panics occurring after PR #184

### ISSUE THREAD

**Issue by kujenga** [gorilla/mux#188]
I just updated the dependencies on a repository to the latest versions, with this package updated to commit `674ef1c28022f33fa7200f59db45e604557f0380`, and we started receiving the following panic (with everything below here being specific to our application):

```
panic: runtime error: slice bounds out of range

goroutine 40 [running]:
panic(0x11f4ee0, 0xc01040e020)
	go/src/runtime/panic.go:443 +0x4e9
github.com/gorilla/mux.getPath(0xc0104c5ea0, 0x0, 0x0)
	github.com/gorilla/mux/mux.go:376 +0x15f
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0107ab860, 0x2a7730fd5f58, 0xc0106a0e00, 0xc0104c5ea0)
	github.com/gorilla/mux/mux.go:80 +0x48
github.com/codegangsta/negroni.Wrap.func1(0x2a7730fd5f58, 0xc0106a0e00, 0xc0104c5ea0, 0xc0107d4000)
```

It appears that this points to the lines introduced in PR: https://github.com/gorilla/mux/pull/184

Specifically, the panic occurs on this line: https://github.com/gorilla/mux/blob/674ef1c28022f33fa7200f59db45e604557f0380/mux.go#L376

Which was introduced here: https://github.com/gorilla/mux/pull/184/files#diff-b303fad270323289aeb17e2f0930ffa7R376

**Comment by kujenga** [comment]
Some additional information here that more tightly scopes this problem, this issue only occurs within the deployed Google App Engine environment. It does not occur in local development with the App Engine SDK.

I have a project here that that I just updated, and demonstrates the same problem (note that I'm using specifically the `ae-glide` branch).  https://github.com/kujenga/vendorexpr/tree/ae-glide

Visiting the URL this app after it's deployed results in the same stack trace:

```
panic: runtime error: slice bounds out of range

goroutine 6 [running]:
panic(0x146efa0, 0xc01040e020)
	go/src/runtime/panic.go:443 +0x4e9
github.com/gorilla/mux.getPath(0xc0104c42a0, 0x0, 0x0)
	github.com/gorilla/mux/mux.go:373 +0x166
github.com/gorilla/mux.(*Router).ServeHTTP(0xc010454460, 0x2af5afbbd818, 0xc010451900, 0xc0104c42a0)
	github.com/gorilla/mux/mux.go:80 +0x48
net/http.(*ServeMux).ServeHTTP(0xc010410ba0, 0x2af5afbbd818, 0xc010451900, 0xc0104c42a0)
	go/src/net/http/server.go:1910 +0x17d
appengine_internal.executeRequestSafely(0xc010451900, 0xc0104c42a0)
	go/src/appengine_internal/api_prod.go:303 +0x8d
appengine_internal.(*server).HandleRequest(0x18af390, 0xc0104dd200, 0xc010400a80, 0xc010439d40, 0x0, 0x0)
	go/src/appengine_internal/api_prod.go:237 +0x15b5
reflect.Value.call(0x149eb00, 0x18af390, 0x213, 0x1555ef8, 0x4, 0xc010513f30, 0x3, 0x3, 0x0, 0x0, ...)
	go/src/reflect/value.go:435 +0x120d
reflect.Value.Call(0x149eb00, 0x18af390, 0x213, 0xc010513f30, 0x3, 0x3, 0
```

**Comment by kujenga** [comment]
For App Engine requests in the deployed environment, The value of `RequestURI` does not contain a "host", so for a request to the root URL on the deployed test app I linked to above, the value of `RequestURI` is `"/"`. 

This is the debug printed output (with `%#v`) of the URL entity:

```go
url: &url.URL{Scheme:"http", Opaque:"", User:(*url.Userinfo)(nil), Host:"vendorexpr-dot-my-app.appspot.com", Path:"/", RawPath:"", RawQuery:"", Fragment:""}
```

I found an issue on Google App Engine relating to this discrepancy:

https://code.google.com/p/googleappengine/issues/detail?id=7679

The fix should be fairly simple, I'll have a PR up shortly.

**Comment by kujenga** [comment]
Opened PR https://github.com/gorilla/mux/pull/189 with a fix and test coverage.

**Comment by elithrar** [comment]
Closed via #189
